docs/bare_metal/c_small_libs.md

```diff
@@ -62,6 +64,29 @@ After you have completed the steps above, add `small` to the `supported_c_libs`
 }
 ```
 
+### Non-returning main()
+
+Arm microlib doesn't support exiting from `main()`; attempting to exit from `main()` causes a bare metal application to crash. Here we show two ways to prevent this.
+
+#### Sleeping in a loop
+
+One recommended technique is to sleep in a loop at the end of `main()`:
+```
+while (true) {
+    sleep();
+}
+```
+
+This is energy-efficient compared to an empty `while (true) {}` loop, which keeps the processor running. A loop is still needed, because `sleep()` returns after the system is woken up by an interrupt.
```

Comment:
```suggestion
This is energy efficient compared to an empty `while (true) {}` loop, which keeps the processor running. A loop is still needed, because `sleep()` returns after the system is woken up by an interrupt.
```